File: JavaScript/easy/palindromeNumber.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    return x.toString().split("").reverse().join() === x
};


//Follow up: Could you solve it without converting the integer to a string?
var isPalindrome = function(x) {
    if(x < 0) return false;
    let reversed  = 0;
     for(let i = x ; i >= 1; i = Math.floor(i/10)) reversed = reversed10 + i%10;
     return reversed === x ;

};

File: Java/easy/palindromeNumber.java
class Solution{
    public boolean isPalindrome(int x){
        if(x < 0 || (x % 10 == 0 && x != 0)){
            return false;
        }

        int reversed = 0;

        while (x > reversed){
            r = r * 10 + x % 10;
            x = x/10;
        }

        return r == x || reversed/10 = x;
    }
}